File: resources/expected/ros2/isolette/lax/src/isolette_cpp_pkg/src/base_code/thermostat_regulate_temperature_manage_regulator_interface_mrit_base_src.cpp
#include "isolette_cpp_pkg/base_headers/thermostat_regulate_temperature_manage_regulator_interface_mrit_base_src.hpp"

//=================================================
//  D O   N O T   E D I T   T H I S   F I L E
//=================================================

thermostat_regulate_temperature_manage_regulator_interface_mrit_base::thermostat_regulate_temperature_manage_regulator_interface_mrit_base() : Node("thermostat_regulate_temperature_manage_regulator_interface_mrit")
{
    cb_group_ = this->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
    subscription_options_.callback_group = cb_group_;

    // Setting up connections
    thermostat_regulate_temperature_manage_regulator_interface_mrit_current_tempWstatus_subscription_ = this->create_subscription<isolette_cpp_pkg_interfaces::msg::TempWstatusimpl>(
        "thermostat_regulate_temperature_manage_regulator_interface_mrit_current_tempWstatus",
        1,
        std::bind(&thermostat_regulate_temperature_manage_regulator_interface_mrit_base::handle_current_tempWstatus, this, std::placeholders::_1), subscription_options_);

    thermostat_regulate_temperature_manage_regulator_interface_mrit_lower_desired_tempWstatus_subscription_ = this->create_subscription<isolette_cpp_pkg_interfaces::msg::TempWstatusimpl>(
        "thermostat_regulate_temperature_manage_regulator_interface_mrit_lower_desired_tempWstatus",
        1,
        std::bind(&thermostat_regulate_temperature_manage_regulator_interface_mrit_base::handle_lower_desired_tempWstatus, this, std::placeholders::_1), subscription_options_);

    thermostat_regulate_temperature_manage_regulator_interface_mrit_upper_desired_tempWstatus_subscription_ = this->create_subscription<isolette_cpp_pkg_interfaces::msg::TempWstatusimpl>(
        "thermostat_regulate_temperature_manage_regulator_interface_mrit_upper_desired_tempWstatus",
        1,
        std::bind(&thermostat_regulate_temperature_manage_regulator_interface_mrit_base::handle_upper_desired_tempWstatus, this, std::placeholders::_1), subscription_options_);

    thermostat_regulate_temperature_manage_regulator_interface_mrit_regulator_mode_subscription_ = this->create_subscription<isolette_cpp_pkg_interfaces::msg::RegulatorMode>(
        "thermostat_regulate_temperature_manage_regulator_interface_mrit_regulator_mode",
        1,
        std::bind(&thermostat_regulate_temperature_manage_regulator_interface_mrit_base::handle_regulator_mode, this, std::placeholders::_1), subscription_options_);

    thermostat_regulate_temperature_manage_regulator_interface_mrit_upper_desired_temp_publisher_ = this->create_publisher<isolette_cpp_pkg_interfaces::msg::Tempimpl>(
        "thermostat_regulate_temperature_manage_heat_source_mhst_upper_desired_temp",
        1);

    thermostat_regulate_temperature_manage_regulator_interface_mrit_lower_desired_temp_publisher_ = this->create_publisher<isolette_cpp_pkg_interfaces::msg::Tempimpl>(
        "thermostat_regulate_temperature_manage_heat_source_mhst_lower_desired_temp",
        1);

    thermostat_regulate_temperature_manage_regulator_interface_mrit_displayed_temp_publisher_ = this->create_publisher<isolette_cpp_pkg_interfaces::msg::Tempimpl>(
        "operator_interface_oip_oit_display_temperature",
        1);

    thermostat_regulate_temperature_manage_regulator_interface_mrit_regulator_status_publisher_ = this->create_publisher<isolette_cpp_pkg_interfaces::msg::Status>(
        "operator_interface_oip_oit_regulator_status",
        1);

    thermostat_regulate_temperature_manage_regulator_interface_mrit_interface_failure_publisher_ = this->create_publisher<isolette_cpp_pkg_interfaces::msg::FailureFlagimpl>(
        "thermostat_regulate_temperature_manage_regulator_mode_mrmt_interface_failure",
        1);

    // timeTriggered callback timer
    periodTimer_ = this->create_wall_timer(std::chrono::milliseconds(1000),
        std::bind(&thermostat_regulate_temperature_manage_regulator_interface_mrit_base::timeTriggered, this), cb_group_);

}

//=================================================
//  C o m m u n i c a t i o n
//=================================================

void thermostat_regulate_temperature_manage_regulator_interface_mrit_base::handle_current_tempWstatus(const isolette_cpp_pkg_interfaces::msg::TempWstatusimpl::SharedPtr msg)
{
    current_tempWstatus_msg_holder = msg;
}

void thermostat_regulate_temperature_manage_regulator_interface_mrit_base::handle_lower_desired_tempWstatus(const isolette_cpp_pkg_interfaces::msg::TempWstatusimpl::SharedPtr msg)
{
    lower_desired_tempWstatus_msg_holder = msg;
}

void thermostat_regulate_temperature_manage_regulator_interface_mrit_base::handle_upper_desired_tempWstatus(const isolette_cpp_pkg_interfaces::msg::TempWstatusimpl::SharedPtr msg)
{
    upper_desired_tempWstatus_msg_holder = msg;
}

void thermostat_regulate_temperature_manage_regulator_interface_mrit_base::handle_regulator_mode(const isolette_cpp_pkg_interfaces::msg::RegulatorMode::SharedPtr msg)
{
    regulator_mode_msg_holder = msg;
}

isolette_cpp_pkg_interfaces::msg::TempWstatusimpl::SharedPtr thermostat_regulate_temperature_manage_regulator_interface_mrit_base::get_current_tempWstatus() {
    return current_tempWstatus_msg_holder;
}

isolette_cpp_pkg_interfaces::msg::TempWstatusimpl::SharedPtr thermostat_regulate_temperature_manage_regulator_interface_mrit_base::get_lower_desired_tempWstatus() {
    return lower_desired_tempWstatus_msg_holder;
}

isolette_cpp_pkg_interfaces::msg::TempWstatusimpl::SharedPtr thermostat_regulate_temperature_manage_regulator_interface_mrit_base::get_upper_desired_tempWstatus() {
    return upper_desired_tempWstatus_msg_holder;
}

isolette_cpp_pkg_interfaces::msg::RegulatorMode::SharedPtr thermostat_regulate_temperature_manage_regulator_interface_mrit_base::get_regulator_mode() {
    return regulator_mode_msg_holder;
}

void thermostat_regulate_temperature_manage_regulator_interface_mrit_base::put_upper_desired_temp(isolette_cpp_pkg_interfaces::msg::Tempimpl msg)
{
    thermostat_regulate_temperature_manage_regulator_interface_mrit_upper_desired_temp_publisher_->publish(msg);
}

void thermostat_regulate_temperature_manage_regulator_interface_mrit_base::put_lower_desired_temp(isolette_cpp_pkg_interfaces::msg::Tempimpl msg)
{
    thermostat_regulate_temperature_manage_regulator_interface_mrit_lower_desired_temp_publisher_->publish(msg);
}

void thermostat_regulate_temperature_manage_regulator_interface_mrit_base::put_displayed_temp(isolette_cpp_pkg_interfaces::msg::Tempimpl msg)
{
    thermostat_regulate_temperature_manage_regulator_interface_mrit_displayed_temp_publisher_->publish(msg);
}

void thermostat_regulate_temperature_manage_regulator_interface_mrit_base::put_regulator_status(isolette_cpp_pkg_interfaces::msg::Status msg)
{
    thermostat_regulate_temperature_manage_regulator_interface_mrit_regulator_status_publisher_->publish(msg);
}

void thermostat_regulate_temperature_manage_regulator_interface_mrit_base::put_interface_failure(isolette_cpp_pkg_interfaces::msg::FailureFlagimpl msg)
{
    thermostat_regulate_temperature_manage_regulator_interface_mrit_interface_failure_publisher_->publish(msg);
}

File: resources/expected/ros2/python-isolette_inverted_topics/lax/src/isolette_py_pkg/isolette_py_pkg/base_code/thermostat_regulate_temperature_manage_regulator_interface_mrit_base_src.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from queue import Queue
from isolette_py_pkg.user_code.thermostat_regulate_temperature_manage_regulator_interface_mrit_src import *
from rclpy.callback_groups import ReentrantCallbackGroup
from isolette_py_pkg_interfaces.msg import TempWstatusimpl
from isolette_py_pkg_interfaces.msg import RegulatorMode
from isolette_py_pkg_interfaces.msg import Tempimpl
from isolette_py_pkg_interfaces.msg import Status
from isolette_py_pkg_interfaces.msg import FailureFlagimpl

#========================================================
# Re-running Codegen will overwrite changes to this file
#========================================================

class thermostat_regulate_temperature_manage_regulator_interface_mrit_base(Node):
    def __init__(self):
        super().__init__("thermostat_regulate_temperature_manage_regulator_interface_mrit")

        self.cb_group_ = ReentrantCallbackGroup()

        # Setting up connections
        self.thermostat_regulate_temperature_manage_regulator_interface_mrit_current_tempWstatus_subscription_ = self.create_subscription(
            TempWstatusimpl,
            "temperature_sensor_cpi_thermostat_current_tempWstatus",
            self.handle_current_tempWstatus,
            1,
            callback_group=self.cb_group_)

        self.thermostat_regulate_temperature_manage_regulator_interface_mrit_lower_desired_tempWstatus_subscription_ = self.create_subscription(
            TempWstatusimpl,
            "operator_interface_oip_oit_lower_desired_tempWstatus",
            self.handle_lower_desired_tempWstatus,
            1,
            callback_group=self.cb_group_)

        self.thermostat_regulate_temperature_manage_regulator_interface_mrit_upper_desired_tempWstatus_subscription_ = self.create_subscription(
            TempWstatusimpl,
            "operator_interface_oip_oit_upper_desired_tempWstatus",
            self.handle_upper_desired_tempWstatus,
            1,
            callback_group=self.cb_group_)

        self.thermostat_regulate_temperature_manage_regulator_interface_mrit_regulator_mode_subscription_ = self.create_subscription(
            RegulatorMode,
            "thermostat_regulate_temperature_manage_regulator_mode_mrmt_regulator_mode",
            self.handle_regulator_mode,
            1,
            callback_group=self.cb_group_)

        self.thermostat_regulate_temperature_manage_regulator_interface_mrit_upper_desired_temp_publisher_ = self.create_publisher(
            Tempimpl,
            "thermostat_regulate_temperature_manage_regulator_interface_mrit_upper_desired_temp",
            1)

        self.thermostat_regulate_temperature_manage_regulator_interface_mrit_lower_desired_temp_publisher_ = self.create_publisher(
            Tempimpl,
            "thermostat_regulate_temperature_manage_regulator_interface_mrit_lower_desired_temp",
            1)

        self.thermostat_regulate_temperature_manage_regulator_interface_mrit_displayed_temp_publisher_ = self.create_publisher(
            Tempimpl,
            "thermostat_regulate_temperature_manage_regulator_interface_mrit_displayed_temp",
            1)

        self.thermostat_regulate_temperature_manage_regulator_interface_mrit_regulator_status_publisher_ = self.create_publisher(
            Status,
            "thermostat_regulate_temperature_manage_regulator_interface_mrit_regulator_status",
            1)

        self.thermostat_regulate_temperature_manage_regulator_interface_mrit_interface_failure_publisher_ = self.create_publisher(
            FailureFlagimpl,
            "thermostat_regulate_temperature_manage_regulator_interface_mrit_interface_failure",
            1)

        # timeTriggered callback timer
        self.periodTimer_ = self.create_timer(1000, self.timeTriggered, callback_group=self.cb_group_)

    def timeTriggered(self):
        pass

    #=================================================
    #  C o m m u n i c a t i o n
    #=================================================

    def handle_current_tempWstatus(self, msg):
        typedMsg = TempWstatusimpl()
        typedMsg.data = msg
        self.current_tempWstatus_msg_holder = typedMsg

    def handle_lower_desired_tempWstatus(self, msg):
        typedMsg = TempWstatusimpl()
        typedMsg.data = msg
        self.lower_desired_tempWstatus_msg_holder = typedMsg

    def handle_upper_desired_tempWstatus(self, msg):
        typedMsg = TempWstatusimpl()
        typedMsg.data = msg
        self.upper_desired_tempWstatus_msg_holder = typedMsg

    def handle_regulator_mode(self, msg):
        typedMsg = RegulatorMode()
        typedMsg.data = msg
        self.regulator_mode_msg_holder = typedMsg

    def get_current_tempWstatus(self):
        return self.current_tempWstatus_msg_holder

    def get_lower_desired_tempWstatus(self):
        return self.lower_desired_tempWstatus_msg_holder

    def get_upper_desired_tempWstatus(self):
        return self.upper_desired_tempWstatus_msg_holder

    def get_regulator_mode(self):
        return self.regulator_mode_msg_holder

    def put_upper_desired_temp(self, msg):
        typedMsg = Tempimpl()
        typedMsg.data = msg
        self.thermostat_regulate_temperature_manage_regulator_interface_mrit_upper_desired_temp_publisher_.publish(typedMsg)

    def put_lower_desired_temp(self, msg):
        typedMsg = Tempimpl()
        typedMsg.data = msg
        self.thermostat_regulate_temperature_manage_regulator_interface_mrit_lower_desired_temp_publisher_.publish(typedMsg)

    def put_displayed_temp(self, msg):
        typedMsg = Tempimpl()
        typedMsg.data = msg
        self.thermostat_regulate_temperature_manage_regulator_interface_mrit_displayed_temp_publisher_.publish(typedMsg)

    def put_regulator_status(self, msg):
        typedMsg = Status()
        typedMsg.data = msg
        self.thermostat_regulate_temperature_manage_regulator_interface_mrit_regulator_status_publisher_.publish(typedMsg)

    def put_interface_failure(self, msg):
        typedMsg = FailureFlagimpl()
        typedMsg.data = msg
        self.thermostat_regulate_temperature_manage_regulator_interface_mrit_interface_failure_publisher_.publish(typedMsg)

